### DISCUSSION TITLE
Connecting to and Arduino with MPF

### DISCUSSION THREAD

**Discussion by brian90254** [missionpinball/missionpinball#93] [Q&A]
Hi folks,
 
This is probably garbage, but I asked ChatGPT to write a config example that would connect to an Arduino, and this is what it gave me:

`hardware:
  platform: mpf_system
  connections:
    - port: /dev/ttyACM0  # Replace with the correct serial port for your Arduino
      baud: 9600
  default_led_boards: arduino

switches:
  s_start_button:
    number: 1
    platform: mpf
    connection: port1
    debounce: 10ms

lights:
  l_red:
    platform: mpf
    connection: arduino
    number: 1
    type: light
  l_green:
    platform: mpf
    connection: arduino
    number: 2
    type: light
  l_yellow:
    platform: mpf
    connection: arduino
    number: 3
    type: light

modes:
  attract:
    start_events: ball_started
    lights:
      l_red:
        color: red
      l_green:
        color: green
      l_yellow:
        color: yellow
    switches:
      s_start_button:
        events_when_activated: start`

So, right off the bat, "platform: mpf_system" doesn't seem to be a real thing, but maybe it should? Or, maybe there's another platform definition that would make this work? Same for "connection: arduino", and so on...

Any ideas? Basically looking for a generic way to have MPF talk to an Arduino. And yes, someone probably has something on GitHub, but looking for a dead-simple example that's easy to read.

Thanks!
B

**Comment by cobra18t** [comment]
The supported hardware platforms are listed here: https://docs.missionpinball.org/hardware/index.html

I am not aware of a generic Arduino platform for MPF. The Arduino Pinball Controller uses the LISY protocol and uses a combo of an Arduino and Raspberry Pi Zero.
